pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -106,12 +107,30 @@ func (r *Reconciler) CheckControlPlaneReady(ctx context.Context, log logr.Logger
 	return clusters.CheckControlPlaneReady(ctx, r.client, log, clusterSpec.Cluster)
 }
 
-// ReconcileWorkerNodes validates the cluster definition and reconciles the worker nodes
-// to the desired state.
+// ReconcileWorkerNodes reconciles the worker nodes to the desired state.
 func (r *Reconciler) ReconcileWorkerNodes(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
-	// Implement reconcile worker nodes here
+	log = log.WithValues("provider", "tinkerbell", "reconcile type", "workers")
+	clusterSpec, err := c.BuildSpec(ctx, clientutil.NewKubeClient(r.client), cluster)
+	if err != nil {
+		return controller.Result{}, errors.Wrap(err, "building cluster Spec for worker node reconcile")
+	}
 
-	return controller.Result{}, nil
+	return controller.NewPhaseRunner().Register(
+		r.ValidateClusterSpec,
+		r.ReconcileWorkers,
+	).Run(ctx, log, clusterSpec)
+}
+
+// ReconcileWorkers applies the worker CAPI objects to the cluster.
+func (r *Reconciler) ReconcileWorkers(ctx context.Context, log logr.Logger, spec *c.Spec) (controller.Result, error) {
```

Comment:
The name of this func has a remarkable similarity to `ReconcileWorkerNodes`. Is there another name we can use?

---

Does this need to be exported?